### PR TITLE
Pull upstream changes

### PR DESCRIPTION
**Pull request by endersstocker**
The upstream repo was updated. We can make further changes that we’ve discussed after this one is merged in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -309,81 +272,245 @@ Please feel free to make pull requests and suggestions.
...
+  def some_function([]), do: :ok
+
+  def some_function([first | rest]) do
+    some_function(rest)
   end
+  ```
 
-  def some_function() do
-    # body omitted
+* <a name="multiple-function-defs"></a>
```
Comment (poops) [inline README.md]: We removed this from ours

File: README.md [hunk 2]
```diff
@@ -309,81 +272,245 @@ Please feel free to make pull requests and suggestions.
...
+* <a name="parentheses-pipe-operator"></a>
+  Use parentheses for one-arity functions when using the pipe operator (`|>`).
+  <sup>[[link](#parentheses-pipe-operator)]</sup>
+
+  ```elixir
+  # not preferred
+  some_string |> String.downcase |> String.strip
+
+  # preferred
+  some_string |> String.downcase() |> String.strip()
```
Comment (poops) [inline README.md]: Preferred way should show multi-line pipes on new lines

File: README.md [hunk 3]
```diff
@@ -309,81 +272,245 @@ Please feel free to make pull requests and suggestions.
...
-    "Hello"
-    |> String.downcase()
-    |> String.strip()
-  another_string <> some_string
+  String.strip(String.downcase(some_string))
 
   # preferred
-  some_string =
-    "Hello"
+  some_string |> String.downcase() |> String.strip()
```
Comment (poops) [inline README.md]: Preferred way should show multi-line pipes on new lines

File: README.md [hunk 4]
```diff
@@ -309,81 +272,245 @@ Please feel free to make pull requests and suggestions.
...
+  String.strip(some_string) |> String.downcase() |> String.codepoints()
 
   # preferred
-  something =
-    if x == 2 do
-      "Hi"
-    else
-      "Bye"
-    end
+  some_string |> String.strip() |> String.downcase() |> String.codepoints()
```
Comment (poops) [inline README.md]: Preferred way should show multi-line pipes on new lines

**Comment by endersstocker** [comment]
@poops Thanks for catching those. Take a look now.

**Pull request merged**